# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

### 1) My Lane and Why

**Chosen Lane:** Lane 2: Refresh / Content Opportunity Scoring

**Rationale:**
Managing a massive content inventory manually is highly inefficient for a content team. By choosing Lane 2, we aim to build an AI-driven decision-support system that acts as an intelligent filtering layer. Based on pre-defined guidelines and observable performance signals, the model will classify and rank pages into distinct operational tiers. 

Low-complexity, clear-opportunity pages can be processed with minimal overhead, while highly complex or uncertain cases—where the AI is prone to errors—will be flagged and routed directly to expert human editors for manual review. This approach optimizes human resource allocation, ensuring that limited operational capacity is focused exclusively on high-impact, high-risk content decisions.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### 2) The Question: Decision, Action, and Cost of a Wrong Call

**The Research Question:**
"Which content pages are demonstrating a sustained decline in performance and should be prioritized for review, and what shared structural or performance signals distinguish successful pages from failing ones?"

**The Decision & Action Policy:**
The core decision is to determine whether a page requires intervention or must be left untouched. 
1. **For High-Performing Pages (Winners):** The policy is 'No Action' to eliminate the risk of disrupting existing traffic.
2. **For Low-Performing Pages (Losers):** The action is a targeted content refresh. This refresh will be guided by an automated analysis of shared success factors extracted from winning pages, while deliberately avoiding the structural traits identified in failing pages.

**The Cost of a Wrong Call:**
* **False Positive (Severe Cost):** Flagging a successful page as 'declining'. If an editor modifies a page that is currently driving significant traffic, it can lead to an immediate, catastrophic loss of search rankings and revenue.
* **False Negative (Opportunity Cost):** Missing a genuinely declining page. This results in continuous traffic bleed and unmitigated revenue erosion over time.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [15]:
import os

# التأكد من وجود مجلد البيانات والملف بداخلة
print("الملفات داخل مجلد raw الحقيقي:")
print(os.listdir("../../data/raw"))

الملفات داخل مجلد raw الحقيقي:
['content_refresh_anonymized.csv']


In [16]:
import duckdb

# قراءة واستكشاف أول 5 صفوف من ملف الـ CSV مباشرة
duckdb.sql("SELECT * FROM '../../data/raw/content_refresh_anonymized.csv' LIMIT 5").show()

┌──────────────────────┬───────────────────┬───────────────┬─────────────┬───────────────────┬────────┬─────────────────┬───────────────┬────────────┬────────────┬───────────────┬────────────────────────┬─────────────────┬────────────┬───────────────┬──────────────┬───────────┬──────────────────────┬─────────────────┬───────────────────┬───────────────────────┬────────────────────┬──────────────────────┬─────────────────┬───────────────────┬──────────────────────┬─────────────────┬───────────────────┬──────────────────┬──────────┬────────────────┬────────────────────────┬────────────────┬─────────────────┬─────────────────┬────────┬──────────────┬─────────────────┬─────────────┬────────────────┬─────────────────┬───────────────┬─────────────────┬───────────┐
│      content_id      │     client_id     │ search_volume │ competition │ competition_level │  cpc   │  content_type   │  main_intent  │ word_count │ char_count │ provider_used │       model_used       │ impressions_90d │ clicks_9

In [17]:
import duckdb

# استعلام لحساب إجمالي الصفحات، والصفحات الهابطة، ومتوسط نسبة الهبوط
query = """
SELECT 
    COUNT(*) as total_pages,
    COUNT(CASE WHEN trend_direction = 'down' THEN 1 END) as declining_pages,
    ROUND(AVG(CASE WHEN trend_direction = 'down' THEN trend_pct END), 2) as avg_decline_percentage
FROM '../../data/raw/content_refresh_anonymized.csv'
"""

duckdb.sql(query).show()

┌─────────────┬─────────────────┬────────────────────────┐
│ total_pages │ declining_pages │ avg_decline_percentage │
│    int64    │      int64      │         double         │
├─────────────┼─────────────────┼────────────────────────┤
│       30000 │           16262 │                 -58.11 │
└─────────────┴─────────────────┴────────────────────────┘



In [18]:
# معرفة أكثر الموديلات المستخدمة في توليد المحتوى الحالي
duckdb.sql("""
    SELECT model_used, COUNT(*) as count 
    FROM '../../data/raw/content_refresh_anonymized.csv' 
    GROUP BY model_used 
    ORDER BY count DESC
""").show()

┌────────────────────────┬───────┐
│       model_used       │ count │
│        varchar         │ int64 │
├────────────────────────┼───────┤
│ gemini-3-flash-preview │ 13271 │
│ NULL                   │  5733 │
│ gpt-4o-mini            │  4981 │
│ gemini-2.5-flash       │  3665 │
│ gpt-5-mini             │  1598 │
│ unknown                │   752 │
└────────────────────────┴───────┘



## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## 4. Careful words: what I can and can't claim

### What this project WILL be able to claim:
* **Decision-Support:** The model acts as an intelligent prioritization tool, giving content teams clear, data-driven recommendations on which of the 16,262 declining pages to refresh first to maximize organic traffic recovery.
* **Directional Trends:** It will accurately flag and categorize historical patterns, identifying which content cohorts are experiencing the steepest drops (e.g., the observed -58.11% average decline).
* **Observed Correlations:** It will highlight strong statistical relationships between content freshness, keyword competition levels, and recent traffic decay.

### What this project WILL NEVER claim:
* **Causal Proof:** It cannot definitively prove that content aging is the *sole* cause of traffic drops, as external factors like structural site issues or aggressive new competitors also play a role.
* **'Predicting Google':** The model does not reverse-engineer or predict Google's search algorithm updates, nor does it guarantee specific ranking positions. It purely optimizes internal resource allocation based on historical performance.

## Self-check

Before you submit, confirm each line honestly:

- [TRUE] Every section above is filled — markdown thinking AND the code that backs it
- [TRUE] The notebook runs top to bottom with no errors (Runtime → Run all)
- [TRUE] No client names, URLs, or private queries anywhere
- [TRUE] My claims use careful words: observed, measured, directional, decision-support
- [TRUE] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.